bcda/auth/plugin/alpha.go

```diff
@@ -68,7 +68,36 @@ func (p *AlphaAuthPlugin) DeleteClient(params []byte) error {
 
 // can treat as a no-op or call RequestAccessToken
 func (p *AlphaAuthPlugin) GenerateClientCredentials(params []byte) ([]byte, error) {
-	return nil, errors.New("not yet implemented")
+	var empty []byte
+	clientID, err := GetParamString(params, "clientID")
+	if err != nil {
+		return nil, err
```

Comment:
Why nil rather than an empty byte slice?

Reply:
Handling was inconsistent throughout the file. It turns out that in go, an empty slice is nil until it is assigned a value, making this a completely pointless variable. It's also not idiomatic go, so I've  replaced these all with nil.